vertx-config-toml/src/main/java/io/vertx/config/toml/TomlProcessor.java

```diff
@@ -0,0 +1,84 @@
+package io.vertx.config.toml;
+
+import com.moandjiezana.toml.Toml;
+import io.vertx.config.spi.ConfigProcessor;
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.json.JsonObject;
+import org.jetbrains.annotations.Contract;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.io.ByteArrayInputStream;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * <p>
+ * A processor for the <a href='https://github.com/toml-lang/toml'>TOML</a>
+ * format.
+ * </p>
+ * <p>
+ * Currently using the
+ * <a href='https://github.com/mwanji/toml4j'>toml4j library</a>. If a newer
+ * version of this library becomes available, you can override the dependency
+ * (assuming the API does not change).
+ * </p>
+ *
+ * @author <a href='lyndon.codes'>Lyndon Armitage</a>
+ */
+public final class TomlProcessor implements ConfigProcessor {
+
+  public TomlProcessor() {
+  }
+
+  @NotNull
+  @Contract(pure = true)
+  @Override
+  public final String name() {
+    return "toml";
+  }
+
+  @Override
+  public final void process(
+    @NotNull Vertx vertx,
+    @Nullable JsonObject configuration,
+    @NotNull Buffer input,
+    @NotNull Handler<AsyncResult<JsonObject>> handler
+  ) {
+    Objects.requireNonNull(vertx, "vertx cannot be null");
+    Objects.requireNonNull(input, "input cannot be null");
+    Objects.requireNonNull(handler, "handler cannot be null");
+
+    if (input.length() == 0) {
+      handler.handle(Future.succeededFuture(new JsonObject()));
+      return;
+    }
+
+    vertx.executeBlocking(
+      future -> {
+        // Currently loads the whole config file into memory
+        try (
+          ByteArrayInputStream inputStream = new ByteArrayInputStream(
+            input.getBytes()
+          )
+        ) {
+
+          Toml toml = new Toml().read(inputStream);
+          Map<String, Object> asMap = Objects.requireNonNull(
```

Comment:
This should be reported to the handler.

Reply:
Right now it is because of the try / catch. But it's a little bit convoluted

Reply:
This was actually a bit of defensive programming on my part as the toml4j library provides the `toMap` method without any documentation. For reference:

```java
  public Map<String, Object> toMap() {
    HashMap<String, Object> valuesCopy = new HashMap<String, Object>(values);
    
    if (defaults != null) {
      for (Map.Entry<String, Object> entry : defaults.values.entrySet()) {
        if (!valuesCopy.containsKey(entry.getKey())) {
          valuesCopy.put(entry.getKey(), entry.getValue());
        }
      }
    }

    return valuesCopy;
  }
```

In the processor would you have preferred something more like:

```java
          Map<String, Object> asMap = toml.toMap();
          if (asMap != null) {
            JsonObject asJson = new JsonObject(asMap);
            future.complete(asJson);
          } else {
            future.fail(
              new NullPointerException("toml library returned a null map")
            );
          }
```

Instead of relying on the catch?